File: mario.c
#include <cs50.h>
#include <stdio.h>


//here I declare the functions I made myself
void pyramid_building(int);
void error_function(void);

int main(void)
{
    printf("Let's build a piramid. Give me a a non-negative integer no greater than 23 for the heigth.\n");

    int levels = get_int();
    if (levels < 24 && levels > -1)
    pyramid_building(levels);
    else
    main();

}

//This is the function that builds the pyramid

int a, b, c, d;
void pyramid_building(int levels)
{
    for (a = 0; a < levels; a++)
    {
        for ( b = 0; b < (levels - (a + 1)); b++)
            printf(" ");
        {
            for (c = 0; c < a+1; c++)
                printf("#");
                printf("  ");
            {
                for (d = 0; d < a+1; d++)
                printf("#");
            }
        }
        printf("\n");
    }
}

File: mario.py
import cs50

def main():
    num_levels = get_pos_int()
    #this is a loop for each level of the pyramid. #the pyramid is written top to bottom. The number of ## printed in a line is the line count, starting from 1
    for level in range(num_levels):
        print(' ' * (num_levels-level+1), end="")
        print("#" * (level+1), end="")
        print('  ', end="")
        print("#" * (level+1), end="")
        print(' ' * (num_levels-level+1), end="")
        print('')


def get_pos_int():
    while True:
        print("Let's build a piramid. Give me a a non-negative integer no greater than 23 for the heigth.")
        n = cs50.get_int()
        if n > 0 or n < 23:
            break
    return n

if __name__ == "__main__":
    main()